osquery/tables/other/augeas.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include <augeas.h>
+#include <boost/algorithm/string/join.hpp>
+
+namespace osquery {
+namespace tables {
+void matchAgueasPattern(QueryData &results,
+                        const augeas *aug,
+                        const std::string &pattern) {
+  char **matches = NULL;
+
+  int len = aug_match(aug, pattern.c_str(), &matches);
+
+  // Handle errors
+  if (!matches)
+    return;
+  else if (len < 0) {
+    const char *error_message = aug_error_message(const_cast<augeas *>(aug));
+    const char *error_message_details =
+        aug_error_details(const_cast<augeas *>(aug));
+    VLOG(1) << "An error has occurred while trying to query augeas: "
+            << error_message;
+
+    if (error_message_details)
```

Comment:
Is it multiline? I thought it just concats the error message.
I think we need to report what went wrong.